File: WordGames/palindrome/palindrome.py
# Program to create a list for all of the words that are palindromes
# Then print the longest palindrome
# importing the word list
from curses import A_ALTCHARSET


sow_pods = open("/home/evergreen/UnderDogsAssignments/sowpods")

# creating lists; one for the words & one for all the words thare palindromes
words = []
palindromes = []
a = 0

# Go through each word in the file and add it to the list, words

for line in sow_pods:
    words.append(line.strip())

# go through words and add any palindrome to the list  palindromes
for a in words:
    reverse_word = a[::-1]
    if a == reverse_word:
        palindromes.append(reverse_word)

print("All the palindromes are " + str(palindromes))

# Find the longest palindrome and print it
longest_palindrome = "AA"
for b in palindromes:
    if len(b) > len(longest_palindrome):
        longest_palindrome = b
print("The longest palindrome is " + longest_palindrome)

File: WordGames/palindrome/palindrome.js
// Program to find all the words containing two consecutive U's
// Not Currently Working
// Importing the fs module
let fs = require("fs")

// Intitializing the readFileLines with the file
const readFileLines = filename =>
   fs.readFileSync(filename)
   .toString('UTF8')
   .split('\n');

// Calling the readFiles function with file name
let words = readFileLines('/home/evergreen/UnderDogsAssignments/sowpods');
let palindromes = []
let longestPalindrome = ['AA']

// Cycle through words and find all the palindromes by comparing last letter for first letters
for (let i = 0; i < words.length - 2; i++) {
    let reverseWord = words[i].split('').reverse().join('');
    if (words[i] === reverseWord) {
        palindromes.push(words[i])
    }}

// Find the longest index in palindrome
for(let j = 0; j < words.length; j++) {
    if (palindromes[j].length > longestPalindrome.length) {
        let longestPalindrome = palindromes[j];
    }}


console.log("All the words that are palindromes are " + palindromes)
console.log("The longest palindrome is " + longestPalindrome)